pkg/steps/release/create_release.go

```diff
@@ -145,6 +121,207 @@ oc adm release extract --from=%q --to=/tmp/artifacts/release-payload-%s
 	return step.Run(ctx, dry)
 }
 
+// importFromReleaseImage uses the provided release image and updates the stable / release streams as
+// appropriate with the contents of the payload so that downstream components are using the correct images.
+// The most common case is to use the correct installer image, tests, and cli commands.
+func (s *assembleReleaseStep) importFromReleaseImage(ctx context.Context, dry bool, providedImage string) error {
```

Comment:
I don't think I fully understand why this is necessary, is this lacking in current version of ci-op or is it just when you want to run a job and specify a specific release?

Reply:
We want to run a ci-operator job where the input to the job is a release image (release qualification tests) or we want to run an upgrade job where we take a previously released image (upgrade tests) where we won't have an image stream to represent them.  The upgrade job:

> Test the upgrade from 4.0.3 to a PR in 4.1 master against the cluster-version-operator component

wants to use the installer image from 4.0.3, so we need to populate `stable-initial` with the contents of 4.0.3 while `stable` continues to be populated from 4.1 master.

Reply:
The other use case is:

> Test installing 4.1.4 and run jobs against it

We don't want to use the image stream for `4.1.4`, we want to grab the payload for `4.1.4` and then use the `installer` image out of the `4.1.4` payload to install the cluster and the `tests` image out of the `4.1.4` payload (in this case) to run the e2e tests.

Reply:
I added a super long godoc to the type, hope that clarifies.